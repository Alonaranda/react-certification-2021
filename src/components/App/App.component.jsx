import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import { Header } from '../../pages/Header/Header';
import { HomeView } from '../../pages/HomeView/HomeView.page';
import { VideoDetails } from '../../pages/VideoDetails/VideoDetails'

//import AuthProvider from '../../providers/Auth';
//import HomePage from '../../pages/Home';
//import LoginPage from '../../pages/Login';
//import NotFound from '../../pages/NotFound';
//import Private from '../Private';
//import Layout from '../Layout';


function App() {

  return (

    <Router>
      <>
        <Header/>
        <Switch>
          <Route exact path ="/" component={HomeView}/>
          <Route exact path="/videoDetails/:videoId" component={VideoDetails}/>
          <Redirect to="/"/>
        </Switch>
      </>
    </Router>



    /*
    <BrowserRouter>
      <AuthProvider>

          <Switch>

            
            <Route exact path="/">
              <Layout>
                <HomePage/>
              </Layout>
            </Route>

            <Route exact path="/login">
              <Layout>
                <LoginPage />
              </Layout>
            </Route>

            <Private exact path="/homeview">
              <HomeView />
            </Private>
            
            <Route path="*">
              <Layout>
                <NotFound />
              </Layout>
            </Route>

          </Switch>

      </AuthProvider>
    </BrowserRouter>*/
  );
  
}

export default App;
