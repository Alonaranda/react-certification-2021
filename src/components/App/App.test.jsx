import React from 'react'
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import App from "./App.component";


describe('Test App component', () => {
    const wrapper = shallow(<App/>)

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have HomeView', () => {
        const header = wrapper.find('HomeView');
        expect(header.length).toEqual(1);
        
    });
});