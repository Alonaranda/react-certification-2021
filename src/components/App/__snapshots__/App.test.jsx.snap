// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should render 1`] = `ShallowWrapper {}`;
